src/vaadin-grid-column.html

```diff
@@ -363,6 +430,18 @@
       }
     }
 
+    /**
+     * @param {string} path
+     * @param {string} header
+     * @param {Element} headerCell
+     * @param {Element} footerCell
+     * @param {Array<Element>} cells
+     * @param {GridBodyRenderer} renderer
+     * @param {GridHeaderFooterRenderer} headerRenderer
+     * @param {HTMLTemplateElement} bodyTemplate
+     * @param {HTMLTemplateElement} headerTemplate
+     * @protected
+     */
     _pathOrHeaderChanged(path, header, headerCell, footerCell, cells, renderer, headerRenderer, bodyTemplate, headerTemplate) {
```

Comment:
Since this is a (complex) observer and none of the defined dependencies have a default value, any of the parameters may be `undefined`.

Looks like the `headerCell` and `footerCell` are only being set at this line https://github.com/vaadin/vaadin-grid/blob/b48c5811a8a19bcf80fd06aa822ad168b1fa263b/src/vaadin-grid.html#L635 and they will always be a `<th>` (header) or `<td>` (footer) element which both are of DOM type [HTMLTableCellElement](https://developer.mozilla.org/en-US/docs/Web/API/HTMLTableCellElement) so we could use this more exact type. Though I think `HTMLElement` would also be fine (Element seems unnecessarily generic here).

`cells` dependency is defined as `_cells.*` in the observer so it will receive a Polymer [change record](https://polymer-library.polymer-project.org/3.0/docs/devguide/observers#deep-observation) object with properties `path`, `value` and `base` so it has an incorrect type. We could add an interface type for the change record object, but it's probably fine to use `{!object | undefined}` here instead.

Here's my suggestion:
```suggestion
    /**
     * @param {string | undefined} path
     * @param {string | undefined} header
     * @param {!HTMLTableCellElement | undefined} headerCell
     * @param {!HTMLTableCellElement | undefined} footerCell
     * @param {!object | undefined} cells
     * @param {GridBodyRenderer | undefined} renderer
     * @param {GridHeaderFooterRenderer | undefined} headerRenderer
     * @param {HTMLTemplateElement | undefined} bodyTemplate
     * @param {HTMLTemplateElement | undefined} headerTemplate
     * @protected
     */
    _pathOrHeaderChanged(path, header, headerCell, footerCell, cells, renderer, headerRenderer, bodyTemplate, headerTemplate) {
```

Reply:
Done.